src/grid/becke.py

```diff
@@ -19,14 +19,53 @@
 # --
 """Becke Weights Module."""
 
+
 import warnings
 
+from grid.utils import get_cov_radii
+
 import numpy as np
 
 
 class BeckeWeights:
     """Becke weights functions holder class."""
 
+    def __init__(self, atom_coors, atom_nums, radii=None, order=3):
```

Comment:
I would suggest to have only radii and order as constructor arguments, which define the method. Another method can be used to apply it on any given molecular geometry and grid.

Reply:
``coors`` is also an attribute in ``molgrid``, I think it can be reused to make this initializer simpler.

Reply:
OK, I agree. I will move `atom_coors` back to `generate_weights` method to make the class general. Do you think the `radii` should become a dictionary? I mean, if `radii` is just a list/array and the order of atoms in the molecule are different, we might use the wrong radius for an atom... Let me know what you think, please.

Reply:
We can put the ``numbers`` back to ``Molgrid`` constructor. This may need some extra structure change.
We may need to add number / radius to atomic grid and extract that information when constructing molgrid. In that case, both radius and coors can be offered directly without manually passing to becke constructor.
In Paul's note, we need radius of the atom to construct proper atomic grid, so it need to be done eventually.

Reply:
That's a good point and probably the best solution. I will differ it to another PR if that is OK with you, as it touches `AtomicGrid`...

Reply:
I will leave this pr open. Once the ``AtomicGrid`` ready, we can merge both.

Reply:
The only caveat I see with what you suggested is that we need to pass `radii` (retrieved from `AtomicGrid` instances) when calling `aim_weights`, which wouldn't be needed for HirshfeldWeights` class... So, for the sake of consistent API, I think we might need to pass `radii` to BeckeWeights constructor... @tczorro & @tovrstra what do you think?

Reply:
Extra args can be dump to ``**kwargs`` and poped in its own`` __call__`` function when needed. But this may not looks as elegant as uniform args. I'm not sure will we add new partition schemes in the future. Those schemes may some need extra args.
What I mentioned above are just some extra concerns. For now, I am fine with either way.

Reply:
The atomic radius is used not only for Becke partitioning, but in general in order to specify the "center" of the radial grids (i.e., the value of the radius that is assigned to x=0 in the original -1 <= x <= +1 grid, or analogously, the point corresponding to the "halfway" gridpoint in Gauss-Laguerre 1D-quadrature.

Reply:
As long as the radii are optional in the constructor of the BeckeWeights class, all is fine. We could support list, array and dictionary, but this can also be changed added later, does not need to be done in this PR. It's up to you. The main point is to fix the mandatory arguments at this stage. 

If the atomic grids would not have a radius parameter, e.g. when using grids from other codes, and when they are not provided as optional argument, one can always fall back to a sensible default.